src/runtime-rs/crates/hypervisor/src/dragonball/inner_device.rs

```diff
@@ -0,0 +1,314 @@
+            }
+            Device::Vfio(_config) => {
+                todo!()
+            }
+            _ => Err(anyhow!("unsupported device {:?}", device)),
+        }
+    }
+
+    fn add_block_device(
+        &mut self,
+        path: &str,
+        id: &str,
+        read_only: bool,
+        no_drop: bool,
+    ) -> Result<()> {
+        let jailed_drive = self.get_resource(path, id).context("get resource")?;
+        self.cached_block_devices.insert(id.to_string());
+
+        let blk_cfg = BlockDeviceConfigInfo {
+            drive_id: id.to_string(),
+            path_on_host: PathBuf::from(jailed_drive),
+            is_direct: self.config.blockdev_info.block_device_cache_direct,
+            no_drop,
+            is_read_only: read_only,
+            ..Default::default()
+        };
+        self.vmm_instance
+            .insert_block_device(blk_cfg)
+            .context("insert block device")
+    }
+
+    fn remove_block_drive(&mut self, id: &str) -> Result<()> {
+        self.vmm_instance
+            .remove_block_device(id)
+            .context("remove block device")?;
+
+        if self.cached_block_devices.contains(id) && self.jailed {
+            self.umount_jail_resource(id)
+                .context("umount jail resource")?;
+            self.cached_block_devices.remove(id);
+        }
+        Ok(())
+    }
+
+    fn add_net_device(&mut self, config: &NetworkConfig) -> Result<()> {
+        let iface_cfg = VirtioNetDeviceConfigInfo {
+            iface_id: config.id.clone(),
+            host_dev_name: config.host_dev_name.clone(),
+            guest_mac: match &config.guest_mac {
+                Some(mac) => MacAddr::from_bytes(&mac.0).ok(),
+                None => None,
+            },
+            ..Default::default()
+        };
+
+        info!(
+            sl!(),
+            "add {} endpoint to {}", iface_cfg.host_dev_name, iface_cfg.iface_id
+        );
+
+        self.vmm_instance
+            .insert_network_device(iface_cfg)
+            .context("insert network device")
+    }
+
+    fn add_vsock(&mut self, config: &VsockConfig) -> Result<()> {
+        let vsock_cfg = VsockDeviceConfigInfo {
+            id: String::from("root"),
+            guest_cid: config.guest_cid,
+            uds_path: Some(config.uds_path.clone()),
+            ..Default::default()
+        };
+
+        self.vmm_instance
+            .insert_vsock(vsock_cfg)
+            .context("insert vsock")
+    }
+
+    fn parse_inline_virtiofs_args(&self, fs_cfg: &mut FsDeviceConfigInfo) -> Result<()> {
+        let mut debug = false;
+        let mut opt_list = String::new();
+
+        fs_cfg.mode = String::from("virtio");
+        fs_cfg.cache_policy = self.config.shared_fs.virtio_fs_cache.clone();
+        fs_cfg.fuse_killpriv_v2 = true;
+
+        // Crate `go_flag` is used to keep compatible with go/flag package.
+        println!("args: {:?}", &self.config.shared_fs.virtio_fs_extra_args);
```

Comment:
`println!` to `info!`

Reply:
Thanks for your review, already updated.